datatypes/src/primitives/feature_data.rs

```diff
@@ -267,7 +261,7 @@ fn null_bitmap_to_bools(null_bitmap: &Option<Bitmap>, len: usize) -> Vec<bool> {
 
 #[derive(Clone, Debug)]
 pub struct CategoricalDataRef<'f> {
-    buffer: arrow::buffer::Buffer,
+    buffer: &'f [u8],
```

Comment:
isn't this worse in term of semantics? is any slice of u8s a valid buffer?

Reply:
I don't think so. I guess we should either decide to remove arrow types here (which I had to because the values() method doesn't provide a buffer anymore) or to just store the reference to the primitive array here and then proceed from there one.
But I guess this is a different refactoring task. The current change just doesn't change anything for that type.